architecture/faust/gui/DaisyPatchInitControlUI.h

```diff
@@ -0,0 +1,230 @@
+#ifndef FAUST_DAISYPATCHINITCONTROL_H
+#define FAUST_DAISYPATCHINITCONTROL_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <string.h>
+
+#include "daisysp.h"
+#include "daisy_seed.h"
+
+#include "faust/gui/DecoratorUI.h"
+#include "faust/gui/ValueConverter.h"
+
+/*******************************************************************************
+ * DaisyPatchInitControlUI : Faust User Interface
+ ******************************************************************************/
+
+class DaisyPatchInitControlUI : public GenericUI
+{
+    
+    private:
+    
+        // Base class for updatable items
+        struct UpdatableZone {
+            FAUSTFLOAT* fZone;
+            
+            UpdatableZone(FAUSTFLOAT* zone) : fZone(zone) {}
+            virtual ~UpdatableZone() {}
+            
+            virtual void update() = 0;
+        };
+    
+        struct SwitchButton : daisy::Switch, UpdatableZone {
+            
+            SwitchButton(FAUSTFLOAT* zone):UpdatableZone(zone)
+            {}
+            
+            void update()
+            {
+                *fZone = RawState();
+            }
+        };
+    
+        // Implement checkbox using daisy::Switch
+        struct CheckButton : daisy::Switch, UpdatableZone {
+            
+            FAUSTFLOAT fLastButton;
+            
+            CheckButton(FAUSTFLOAT* zone):UpdatableZone(zone), fLastButton(0)
+            {}
+            
+            void update()
+            {
+                FAUSTFLOAT button = RawState();
+                if (button == 1.0 && (button != fLastButton)) {
+                    *fZone = !*fZone;
+                }
+                fLastButton = button;
+            }
+        };
+   
+        // Implements an analog knob by calling Process on the knob instance
+        // that has already been created inside DaisyPatchSM
+        struct AnalogKnob : UpdatableZone {
+           
+            std::unique_ptr<ValueConverter> fConverter;
+            daisy::AnalogControl* fControl;
+
+            FAUSTFLOAT fStep;
+            bool fQuantizeToStep;
+            
+            AnalogKnob(daisy::AnalogControl* control, FAUSTFLOAT* zone, std::unique_ptr<ValueConverter>& converter, int rate, FAUSTFLOAT step, bool quantize)
+            : UpdatableZone(zone), fConverter(std::move(converter)), fControl(control), fStep(step), fQuantizeToStep(quantize)
+            {
+            }
+            
+            void update()
+            {
+                FAUSTFLOAT newValue = fConverter->ui2faust(fControl->Process());
```

Comment:
Unlike inside `DaisyControlUI`, no ADC related calls are here, the `DaisyPatchSM` class exposes these "control" instances already.

Much of the other code is the same and could be shared between them though.